widgetastic_manageiq/__init__.py

```diff
@@ -3437,17 +3437,19 @@ def all_entity_names(self):
         # get_all uses self.entity_names, which handles versioned name query
         return [e.name for e in self.get_all(surf_pages=True)]
 
-    def get_all(self, surf_pages=False):
+    def get_all(self, surf_pages=False, slice=None):
         """ obtains all entities like QuadIcon displayed by view
         Args:
             surf_pages (bool): current page entities if False, all entities otherwise
+            slice(slice object): if None, return all elements, else apply {slice} to slice the
+            elements returned.
 
-        Returns: all entities (QuadIcon/etc.) displayed by view
+        Returns: all entities (QuadIcon/etc.) displayed by view, or sliced entities if slice.
         """
         if not surf_pages:
             return [
                 self.parent.entity_class(parent=self, entity_id=el["entity_id"], name=el["name"])
-                for el in self._current_page_elements
+                for el in self._current_page_elements[slice]
```

Comment:
Has this been tested with the default of None? `list[None]` is a TypeError

Reply:
I really thought I tested this, but I guess not. I thought PRT would run on existing tests. This would fail for all tests using entities.get_all().

Reply:
I know I ran this without slice.